go/libkb/assertion.go

```diff
@@ -202,31 +232,67 @@ type AssertionHTTPS struct{ AssertionURLBase }
 type AssertionDNS struct{ AssertionURLBase }
 type AssertionFingerprint struct{ AssertionURLBase }
 
-func (b AssertionURLBase) Check() error {
+func (a AssertionHTTP) CheckAndNormalize() (AssertionURL, error) {
```

Comment:
Any reason not to define this on `AssertionURLBase` once instead of four separate identical times?